### PR TITLE
Expose Journal-related files to allow external Journal implementation integration.

### PR DESCRIPTION
**Pull request by ifesdjeen**
CASSANDRA-20112

### CODE REVIEW

File: accord-core/src/main/java/accord/impl/AbstractLoader.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+                case TRUNCATE_WITH_OUTCOME:
+                case TRUNCATE:
+                case ERASE:
+                    command = Commands.purge(command, command.participants(), cleanup);
+            }
+        }
+
+        command = safeStore.unsafeGet(txnId).update(safeStore, command);
+        if (command.status() == Truncated)
+            safeStore.progressLog().clear(command.txnId());
```
Comment (belliottsmith) [inline accord-core/src/main/java/accord/impl/AbstractLoader.java]: is this something we need for both implementations? I'm not sure, it looks like something hacky we would do for the in memory implementation, that perhaps should be done more cleanly (perhaps tied to `Cleanup` invocation). But also perhaps this is fine.

File: accord-core/src/main/java/accord/impl/InMemoryCommandStore.java [hunk 2]
```diff
@@ -1377,99 +1375,86 @@ public void clearForTesting()
...
 
+        @Override
+        public void load(Command command, Journal.OnDone onDone)
+        {
+            try
+            {
+                commandStore.executeInContext(commandStore,
+                                              context(command, ASYNC),
+                                              safeStore -> loadInternal(command, safeStore));
+                onDone.success();
```
Comment (belliottsmith) [inline accord-core/src/main/java/accord/impl/InMemoryCommandStore.java]: not super important, but perhaps this should be invoked outside of the try block so we don't invoke onDone.failure(t) if success callback fails

File: accord-core/src/test/java/accord/burn/BurnTest.java [hunk 3]
```diff
@@ -18,675 +18,13 @@
...
-
-    private static Verifier createVerifier(int keyCount)
-    {
-        if (!ElleVerifier.Support.allowed())
-            return new StrictSerializabilityVerifier("", keyCount);
-        return CompositeVerifier.create(new StrictSerializabilityVerifier("", keyCount),
-                                        new ElleVerifier());
-    }
-
-    public static void main(String[] args)
```
Comment (belliottsmith) [inline accord-core/src/test/java/accord/burn/BurnTest.java:531]: Should the main method stay here perhaps?

File: accord-core/src/main/java/accord/impl/AbstractLoader.java (hunk 1 above)
Comment (ifesdjeen) [reply accord-core/src/main/java/accord/impl/AbstractLoader.java]: Double checking.

File: accord-core/src/test/java/accord/impl/basic/Cluster.java [hunk 4]
```diff
@@ -778,7 +780,7 @@ private static void verifyConsistentRestore(Int2ObjectHashMap<NavigableMap<TxnId
                     if (!store.unsafeCommands().containsKey(txnId))
                     {
                         Command beforeCommand = before.get(txnId);
-                        if (beforeCommand.saveStatus() == SaveStatus.Erased)
```
Comment (belliottsmith) [inline accord-core/src/test/java/accord/impl/basic/Cluster.java]: Should we permit this? I don't think we should even *have* any commands where `saveStatus` is `Uninitialised` - might indicate some other problem? If the command is `Uninitialised` it means we didn't touch it, so we shouldn't have saved it.

File: accord-core/src/test/java/accord/impl/basic/DelayedCommandStores.java [hunk 5]
```diff
@@ -153,7 +155,8 @@ public void validateRead(Command current)
                 return;
 
             // Journal will not have result persisted. This part is here for test purposes and ensuring that we have strict object equality.
-            Command reconstructed = journal.reconstruct(id, current.txnId());
+            // TODO: redundant/durable before
```
Comment (belliottsmith) [inline accord-core/src/test/java/accord/impl/basic/DelayedCommandStores.java]: This TODO looks important

File: accord-core/src/test/java/accord/impl/mock/MockConfigurationService.java [hunk 6]
```diff
@@ -18,27 +18,27 @@
...
-import org.junit.jupiter.api.Assertions;
-
 import java.util.*;
 
 public class MockConfigurationService implements TestableConfigurationService
 {
     private final MessageSink messageSink;
     private final List<Topology> epochs = new ArrayList<>();
-    private final Map<Long, EpochReady> acks = new HashMap<>();
+    private final Map<Long, ConfigurationService.EpochReady> acks = new HashMap<>();
```
Comment (belliottsmith) [inline accord-core/src/test/java/accord/impl/mock/MockConfigurationService.java]: static import class?

File: accord-core/src/test/java/accord/impl/mock/MockConfigurationService.java [hunk 7]
```diff
@@ -18,27 +18,27 @@
...
     private final List<AsyncResult<Void>> syncs = new ArrayList<>();
-    private final List<Listener> listeners = new ArrayList<>();
+    private final List<ConfigurationService.Listener> listeners = new ArrayList<>();
```
Comment (belliottsmith) [inline accord-core/src/test/java/accord/impl/mock/MockConfigurationService.java]: static import class?

**Review by belliottsmith — APPROVED** [review]
Left some minor feedback, otherwise LGTM +1

File: accord-core/src/test/java/accord/impl/mock/MockConfigurationService.java (hunk 7 above)
Comment (ifesdjeen) [reply accord-core/src/test/java/accord/impl/mock/MockConfigurationService.java]: I will double check on commit / when sequashing, I think these are artifact of moving files back and forth.

File: accord-core/src/test/java/accord/impl/basic/DelayedCommandStores.java (hunk 5 above)
Comment (ifesdjeen) [reply accord-core/src/test/java/accord/impl/basic/DelayedCommandStores.java]: Fixed this, you are right, should not have left it for TODO.

File: accord-core/src/test/java/accord/burn/BurnTest.java (hunk 3 above)
Comment (ifesdjeen) [reply accord-core/src/test/java/accord/burn/BurnTest.java:531]: Moved back.

File: accord-core/src/test/java/accord/impl/mock/MockConfigurationService.java (hunk 6 above)
Comment (ifesdjeen) [reply accord-core/src/test/java/accord/impl/mock/MockConfigurationService.java]: Reverted all changes that are not related directly to the patch, including this one.

File: accord-core/src/test/java/accord/impl/basic/Cluster.java (hunk 4 above)
Comment (ifesdjeen) [reply accord-core/src/test/java/accord/impl/basic/Cluster.java]: Agreed, this should not be the case; fixed now

File: accord-core/src/main/java/accord/impl/AbstractLoader.java (hunk 1 above)
Comment (ifesdjeen) [reply accord-core/src/main/java/accord/impl/AbstractLoader.java]: Removed and ran several seeds with 10k operations, which are passing.

File: accord-core/src/main/java/accord/impl/InMemoryCommandStore.java (hunk 2 above)
Comment (ifesdjeen) [reply accord-core/src/main/java/accord/impl/InMemoryCommandStore.java]: did that in both methods

**Comment by ifesdjeen** [comment]
Committed manually

**Pull request closed without merging**